src/animals/animals.service.ts

```diff
@@ -1,15 +1,162 @@
-import { Injectable } from '@nestjs/common';
+import {
+  BadRequestException,
+  Injectable,
+  InternalServerErrorException,
+  NotFoundException,
+} from '@nestjs/common';
 import { InjectModel } from '@nestjs/mongoose';
-import { Model } from 'mongoose';
+import { Model, Types } from 'mongoose';
+import { RedisCacheService } from 'src/redis-cache/redis-cache.service';
+import { AnimalTypeParam } from 'src/validations/type.validator';
+import { AnimalDto, UpdateAnimalDto, AnimalNameDto } from './dto/animal.dto';
 import { Animal, AnimalDocument } from './schemas/animal.schema';
 
 @Injectable()
 export class AnimalsService {
   constructor(
     @InjectModel(Animal.name) private animalModel: Model<AnimalDocument>,
+    private readonly redisCacheManager: RedisCacheService,
   ) {}
 
-  async findAll() {
-    return this.animalModel.find();
+  async findAll(): Promise<Animal[]> {
+    try {
+      return await this.animalModel.find();
+    } catch (err) {
+      // console.error(err);
+      throw new InternalServerErrorException();
```

Comment:
tutaj jakies customowe errory mozesz rzucac...

Reply:
Okej, to tutaj jeszcze poprawię.

Reply:
custoowy wyjatek. wywal console loga